index.js

```diff
@@ -74,8 +81,59 @@ router.post('/pointers/:name', async ctx => {
   ctx.body = pointer
 })
 
+// Update revshare pointer
+router.put('/pointers/:name', auth.authorize, async ctx => {
```

Comment:
there's a lot of code duplication between this function and `POST /pointers/:name`. We could factor it out into something like `setPointerDetails` and call it only if the pointer exists (if the route is `PUT /pointers/:name`) or call it only if the pointer does not exist (if the route is `POST /pointers/:name`)